### PR TITLE
Matt's Python edits and moves for IA restructure

### LINKED ISSUE #2078: Create new page: Models
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

dbt Models - https://docs.getdbt.com/docs/building-a-dbt-project/building-models 

### What part(s) of the page would you like to see updated?

- Create a new "Models" page such that it is a What is a dbt model? overview page. 
- Move the overview content from the "dbt Models" page over to this new page. The "dbt Models" page will be renamed to "SQL models". 

### Additional information

_No response_

### LINKED ISSUE #2092: Rewrite the "dbt Models" page
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

dbt Models -- https://docs.getdbt.com/docs/building-a-dbt-project/building-models

### What part(s) of the page would you like to see updated?

- Rewrite this page so it's more of an "SQL models" page 
- Add missing information about the config property file. See [slack thread](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1664225591214759) for details.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
## Checklist
If you added new pages (delete if not applicable):
- [ x] The page has been added to `website/sidebars.js`
- [ x] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ x] The page has been removed from `website/sidebars.js`
- [ x] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/build/models.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: "About dbt models"
+id: "models"
+---
+
+## Overview
+
+dbt Core and Cloud are composed of different moving parts working harmoniously. All of them are important to what dbt does — transforming data—the 'T' in ELT. When you execute `dbt run`, you are running a model that will transform your data without that data ever leaving your warehouse.
+
+Models are where your developers spend most of their time within a dbt environment. Models are primarily written as a `select` statement and saved as a `.sql` file. While the definition is straightforward, the complexity of the execution will vary from environment to environment.  Models will be written and rewritten as needs evolve and your organization finds new ways to maximize efficiency.
```
Comment (runleonarun) [inline website/docs/docs/build/models.md:10]:
```suggestion
Models are where developers spend most of their time within a dbt environment. Models are primarily written as a `select` statement and saved as a `.sql` file. While the definition is straightforward, the complexity of the execution will vary from environment to environment.  Models will be written and rewritten as needs evolve and your organization finds new ways to maximize efficiency.
```

File: website/docs/docs/build/models.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+SQL is the language most dbt users will utilize, but it is not the only one for building models. Starting in version 1.3, dbt Core and dbt Cloud support Python models. These are useful for training or deploying data science models, complex transformations, or where a specific Python package meets a need.
```
Comment (runleonarun) [inline website/docs/docs/build/models.md]: @matthewshaver I'm having trouble with this sentence and wasn't sure how to fix it. Maybe change these > Python models. This part feels like a new sentence? "where a specific Python package meets a need"
> These are useful for training or deploying data science models, complex transformations, or where a specific Python package meets a need.

File: website/docs/docs/build/python-models.md [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+- [Configuring Python Models](#configuring-python-models)
+- [Python Specific Functionality](#configuring-python-models)
+- [Limitations](#limitations)
+- [Supported data platforms](#supported-data-platforms)
+
+:::info
+
+dbt Core v1.3 adds support for Python models. Note that only [specific data platforms](#specific-data-platforms) support dbt-py models.
+
+We encourage you to:
```
Comment (runleonarun) [inline website/docs/docs/build/python-models.md:13]: Can we move this out of the note? It's a huge and hard-to-read this way. 

I also suggests versioning this paragraph for 1.3 so it's less confusing:
> In the following article, you'll see a section titled "❓ dbt questions." We are excited to release the first set of functionality in v1.3, which will solve real use cases. We also know this is the first step toward a much wider field of possibility. We don't pretend to have all the answers.

File: website/docs/docs/build/python-models.md [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+
+## Overview
+
+dbt Python ("dbt-py") models will help you solve use cases that can't be solved with SQL. You can perform analyses using tools available in the open-source Python ecosystem, including state-of-the-art packages for data science and statistics. Before, you would have needed separate infrastructure and orchestration to run Python transformations in production. Python transformations defined in dbt are models in your project with all the same capabilities around testing, documentation, and lineage.
+
+<VersionBlock firstVersion="1.0" lastVersion="1.2">
+
+Python models are supported in dbt Core 1.3 and higher.  Learn more about [upgrading your version in dbt Cloud](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/cloud-upgrading-dbt-versions) and [upgrading dbt Core versions](https://docs.getdbt.com/docs/core-versions#upgrading-to-new-patch-versions).
+
+To read more about Python models, change the docs version to 1.3 (or higher) in the menu bar.
```
Comment (runleonarun) [inline website/docs/docs/build/python-models.md]: This sentence feels like it needs a link to the 1.3 version. Hopefully this works in markdown:
```suggestion
To read more about Python models, change the [docs version to 1.3](/docs/building-a-dbt-project/building-models/python-models?version=1.3) (or higher) in the menu bar.
```

**Review by runleonarun — CHANGES_REQUESTED** [review]
@matthewshaver I had a few minor styling suggestions but this looks SO GOOD! ❤️

File: website/docs/docs/build/python-models.md (hunk 3 above)
Comment (matthewshaver) [reply website/docs/docs/build/python-models.md:13]: YES!  Thank you for that feedback.  I wasn't a fan.

File: website/docs/docs/build/models.md (hunk 2 above)
Comment (matthewshaver) [inline website/docs/docs/build/models.md]:
```suggestion
SQL is the language most dbt users will utilize, but it is not the only one for building models. Starting in version 1.3, dbt Core and dbt Cloud support Python models. Python models are useful for training or deploying data science models, complex transformations, or where a specific Python package meets a need &mdash; such as using the `dateutil` library to parse dates.
```

Comment (matthewshaver) [reply website/docs/docs/build/models.md]: Got more information to expand from Doug Beatty.  Think it feels a little better now

**Review by runleonarun — APPROVED** [review]
LGTM

File: website/docs/docs/build/python-models.md [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+
+## Overview
+
+dbt Python ("dbt-py") models will help you solve use cases that can't be solved with SQL. You can perform analyses using tools available in the open-source Python ecosystem, including state-of-the-art packages for data science and statistics. Before, you would have needed separate infrastructure and orchestration to run Python transformations in production. Python transformations defined in dbt are models in your project with all the same capabilities around testing, documentation, and lineage.
+
+<VersionBlock firstVersion="1.0" lastVersion="1.2">
+
+Python models are supported in dbt Core 1.3 and higher.  Learn more about [upgrading your version in dbt Cloud](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/cloud-upgrading-dbt-versions) and [upgrading dbt Core versions](https://docs.getdbt.com/docs/core-versions#upgrading-to-new-patch-versions).
+
+To read more about Python models, change the [docs version to 1.3](/docs/building-a-dbt-project/building-models/python-models?version=1.3) (or higher) in the menu bar.
```
Comment (runleonarun) [inline website/docs/docs/build/python-models.md]:
```suggestion
To read more about Python models, change the [docs version to 1.3](/docs/build/python-models?version=1.3) (or higher) in the menu bar.
```

**Pull request merged**